code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -69,22 +69,41 @@
 	text_dehack = "You reboot [name] and restore the sound control system."
 	text_dehack_fail = "[name] refuses to accept your authority!"
 
-/mob/living/simple_animal/bot/honkbot/get_controls(mob/user)
-	var/dat
-	dat += hack(user)
-	dat += showpai(user)
-	dat += text({"
-	<TT><B>Honkomatic Bike Horn Unit v1.0.7 controls</B></TT><BR><BR>
-	Status: []<BR>
-	Behaviour controls are [locked ? "locked" : "unlocked"]<BR>
-	Maintenance panel is [open ? "opened" : "closed"]<BR>"},
-
-	"<A href='?src=[UID()];power=1'>[on ? "On" : "Off"]</A>")
-
-	if(!locked || issilicon(user) || user.can_admin_interact())
-		dat += "Auto Patrol <A href='?src=[UID()];operation=patrol'>[auto_patrol ? "On" : "Off"]</A><BR>"
-
-	return	dat
+/mob/living/simple_animal/bot/honkbot/show_controls(mob/user)
+	ui_interact(user)
+
+/mob/living/simple_animal/bot/honkbot/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "BotHonk", name, 500, 520)
+		ui.open()
+
+/mob/living/simple_animal/bot/honkbot/ui_data(mob/user)
+	var/list/data = ..()
+	return data
+
```

Comment:
redundant proc override. Just remove it

Reply:
it pulls the autopatrol, on off data from the parent, it is still used.